emails/tests/mgmt_process_emails_from_sqs_tests.py

```diff
@@ -284,62 +361,124 @@ def test_no_body_deleted(mock_sqs_client, caplog, test_settings):
-    temp_error = make_client_error(
-        "Email sending has been disabled.", "AccountSendingPausedException"
-    )
-    mock_sns_inbound_logic.side_effect = (temp_error, temp_error)
+def test_ses_generic_failure(
+    mock_sns_inbound_logic: Mock, mock_sqs_client: Mock, caplog: LogCaptureFixture
+) -> None:
+    """The command does not retry generic SES failures."""
+    internal_error = make_client_error(code="InternalError")
+    mock_sns_inbound_logic.side_effect = (internal_error, None)
     msg = fake_sqs_message(json.dumps(TEST_SNS_MESSAGE))
     mock_sqs_client.return_value = fake_queue([msg], [])
     call_command(COMMAND_NAME)
     summary = summary_from_exit_log(caplog)
     assert summary["total_messages"] == 1
     assert summary["failed_messages"] == 1
-    assert summary["pause_count"] == 1
     msg.delete.assert_not_called()
 
 
-def test_ses_generic_failure(mock_sns_inbound_logic, mock_sqs_client, caplog):
-    """The command does not retry generic SES failures."""
-    internal_error = make_client_error(code="InternalError")
-    mock_sns_inbound_logic.side_effect = (internal_error, None)
+def test_ses_python_error(
+    mock_sns_inbound_logic: Mock,
+    mock_sqs_client: Mock,
+    test_settings: SettingsWrapper,
+    caplog: LogCaptureFixture,
+) -> None:
+    """The command catches processing failures."""
+    test_settings.PROCESS_EMAIL_MAX_SECONDS = 2
+    mock_sns_inbound_logic.side_effect = ValueError("bad stuff")
     msg = fake_sqs_message(json.dumps(TEST_SNS_MESSAGE))
     mock_sqs_client.return_value = fake_queue([msg], [])
     call_command(COMMAND_NAME)
     summary = summary_from_exit_log(caplog)
     assert summary["total_messages"] == 1
     assert summary["failed_messages"] == 1
     msg.delete.assert_not_called()
+    rec2 = caplog.records[1]
+    assert rec2.msg == "Message processed"
+    rec2_extra = log_extra(rec2)
+    assert rec2_extra["success"] is False
+    assert rec2_extra["error_type"] == "ValueError"
+    assert rec2_extra["error"] == "bad stuff"
+
+
+def test_ses_slow(
+    mock_sns_inbound_logic: Mock,
+    mock_sqs_client: Mock,
+    mock_process_pool_future: Mock,
+    test_settings: SettingsWrapper,
+    caplog: LogCaptureFixture,
+) -> None:
+    test_settings.PROCESS_EMAIL_MAX_SECONDS_PER_MESSAGE = 120
+    msg = fake_sqs_message(json.dumps(TEST_SNS_MESSAGE))
+    mock_sqs_client.return_value = fake_queue([msg], [], [])
+    mock_process_pool_future._is_stalled.side_effect = [False, False, True]
+    call_command(COMMAND_NAME)
+    summary = summary_from_exit_log(caplog)
+    assert summary["total_messages"] == 1
+    msg.delete.assert_called()
+    rec2 = caplog.records[1]
+    assert rec2.msg == "Message processed"
+    rec2_extra = log_extra(rec2)
+    assert rec2_extra["success"] is True
+    assert rec2_extra["message_process_time_s"] < 120.0
+    assert rec2_extra["subprocess_setup_time_s"] == 1.0
+    assert mock_process_pool_future._timeouts == [1.0]
```

Comment:
Good question! I got the mock backwards. This line:

`mock_process_pool_future._is_stalled.side_effect = [False, False, True]`

should be:

`mock_process_pool_future._is_stalled.side_effect = [True, True, False]`

`_timeouts` is `[1.0]` because it was called one. When you call `future.wait(1.0)` on the mock future, it runs:

```python
            def call_wait(timeout: float) -> None:
                mocked_clocks(timeout)
                mock_future._timeouts.append(timeout)
                if not mock_future._is_stalled():
                    mock_future._ready = True
                    try:
                        ret = func(*args)
                    except BaseException as e:
                        if error_callback:
                            error_callback(e)
                    else:
                        if callback:
                            callback(ret)
```

So, `_timeouts` *always* has a value when `future.wait()` is called.